### PR TITLE
readded query for field names as optional search condition; added fie…

### PR DESCRIPTION
**Pull request by remi128**
…ld validation

#77

### CODE REVIEW

File: vertx-pojo-mapper-common-test/src/main/java/de/braintags/vertx/jomnigate/testdatastore/mapper/typehandler/BaseRecord.java [hunk 1]
```diff
@@ -241,6 +243,20 @@ private boolean compareNumber(Object value, Object compareValue, String fieldNam
   private boolean compareArray(Object value, Object compareValue, String fieldName) {
     if (!compareValue.getClass().isArray())
       throw new MappingException("Contents are not equal: " + fieldName);
+    int l1 = Array.getLength(value);
+    int l2 = Array.getLength(compareValue);
+    if (l1 != l2) {
+      throw new MappingException("different lenght in arrays: " + l1 + " / " + l2);
+    }
+    // first clear wether all elements are existing, equal the order
```
Comment (sschmittBt) [inline vertx-pojo-mapper-common-test/src/main/java/de/braintags/vertx/jomnigate/testdatastore/mapper/typehandler/BaseRecord.java:251]: I don't really understand what you are trying to say here.
But regardless, couldn't you just use "Arrays.equals"?

**Review by sschmittBt — COMMENTED** [review]
With this implementation, there are no checks or restrictions to prevent someone from using just Strings for search fields. 
This way, we are back to not knowing what fields are queried, and we are not able to create indices automatically for each field that is searched.

I understand that you don't want to remove the ability to use String fields, but there should be away to configure the system to disallow it. Or restrict the scope where you can create those queries, so it will be the exception instead of an equal possibiltiy next to IndexedFields

File: vertx-pojo-mapper-common-test/src/main/java/de/braintags/vertx/jomnigate/testdatastore/mapper/typehandler/BaseRecord.java (hunk 1 above)
Comment (remi128) [reply vertx-pojo-mapper-common-test/src/main/java/de/braintags/vertx/jomnigate/testdatastore/mapper/typehandler/BaseRecord.java:251]: just a special check to guarantee the detailed information wether an array is different in length, contains all elements of the other one and is sorted correctly in this order. Arrays.equals wouldn't give the second part.

Comment (remi128) [reply vertx-pojo-mapper-common-test/src/main/java/de/braintags/vertx/jomnigate/testdatastore/mapper/typehandler/BaseRecord.java:251]: This part is only to reproduce the existing ability to search by name. Additionally will come #79 as refactoring job, where inside we can define the correct behaviour of both systems, where an administratable factory system should be defined.

**Pull request merged**